soql-standalone-parser/src/main/scala/com/socrata/soql/ast/Select.scala

```diff
@@ -423,6 +430,20 @@ case class Select(
   def toStringWithFrom(fromTable: TableName): String = {
     docWithFrom(Some(fromTable)).layoutSmart(LayoutOptions(pageWidth = PageWidth.Unbounded)).toString
   }
+
+  def validate(): Select = {
+    distinct match {
+      case DistinctOn(exprs) if orderBys.nonEmpty =>
+        val orderBySet = orderBys.map(_.expression).toSet
+        val distinctNotInOrderBy = (exprs.toSet -- orderBySet)
+        if (distinctNotInOrderBy.nonEmpty) {
+          throw new BadParse("SELECT DISTINCT ON expressions must match initial ORDER BY expressions", distinctNotInOrderBy.head.position)
```

Comment:
This isn't really a _parse_ error per se but that's fine for now I guess.  But I think is too loose - it should probably be `orderBys.take(exprs.size).map etcetc` to get the "initial" part of the condition right.